10-gis.Rmd

```diff
@@ -252,22 +255,35 @@ Similarly to the previous step, we should start by looking at this algorithm's h
 qgis_show_help("grass7:v.clean")
 ```
 
-You may notice that the help text is quite long and contains a lot of arguments.^[Also note that these arguments, contrary to the QGIS's ones, are in lower case.]
+We have omitted the output here, because the help text is quite long and contains a lot of arguments.^[Also note that these arguments, contrary to the QGIS's ones, are in lower case.]
 This is because `v.clean` is a multi tool -- it can clean different types of geometries and solve different types of topological problems.
 For this example, let's focus on just a few arguments, however, we encourage you to visit [this algorithm's documentation](https://grass.osgeo.org/grass78/manuals/v.clean.html) to learn more about `v.clean` capabilities.
 
+```{r, eval=FALSE}
+qgis_arguments("grass7:v.clean") |>
+  dplyr::select(name, description) |>
+  dplyr::slice_head(n = 4)
+#> # A tibble: 4 × 2
+#>   name      description                              
+#>   <chr>     <chr>                                    
+#> 1 input     Layer to clean                           
+#> 2 type      Input feature type                       
+#> 3 tool      Cleaning tool                            
+#> 4 threshold Threshold (comma separated for each tool)
+```
+
 The main argument for this algorithm is `input` -- our vector object.
 Next, we need to select a tool -- a cleaning method. ^[It is also possible to select several tools, which will then be executed sequentially.]
 About a dozen tools exist in `v.clean` allowing to remove duplicate geometries, remove small angles between lines, or remove small areas, among others.
-In this case, we are interested in the latter tool, `rmarea`, which is identified by the number 10.
-Several of the tools, `rmarea` included, expect an additional argument `threshold`, which behavior depends on the selected tool.
+In this case, we are interested in the latter tool, `rmarea`.
+Several of the tools, `rmarea` included, expect an additional argument `threshold`, whose behavior depends on the selected tool.
 In our case, the `rmarea` tool removes all areas smaller or equal to a provided `threshold`. 
 
 Let's run this algorithm and convert its output into a new `sf` object `clean_sf`.
 
 ```{r 09-gis-7c, eval=FALSE}
 clean = qgis_run_algorithm("grass7:v.clean", input = union_sf,
-                           tool = 10, threshold = 25000, .quiet = TRUE)
+                           tool = "rmarea", threshold = 25000, .quiet = TRUE)
```

Comment:
:+1:

Reply:
Nice. I did not know that I can use the name instead of the number.